tmt/steps/prepare/distgit.py

```diff
@@ -246,6 +247,31 @@ def go(
         except tmt.utils.RunError as error:
             raise tmt.utils.PrepareError("Unable to 'rpmbuild -bp'.", causes=[error])
 
+        # Workaround around new rpm behavior, https://github.com/teemtee/tmt/issues/2987
+        # No hardcoded name, should keep working in the future
+        cmd = Command(
+            "rpmbuild",
+            "-bc",
+            "--short-circuit",
+            "--nodeps",
+            "--define",
+            '__spec_build_pre echo XYZbuilddir=%{_builddir}; exit 0',
+            spec_name,
+            *dir_defines)
+        outcome = guest.execute(command=cmd, cwd=source_dir).stdout or ''
+        match = re.search(r'XYZbuilddir=(.+)', outcome)
+        builddir = Path(match.group(1)) if match else None
+
+        # But if the %build is missing in spec (e.g. in our test) the previous output was empty
+        if builddir is None:
+            guest.execute(command=ShellScript(
+                "shopt -s dotglob; if test -e */SPECPARTS; then mv ./*-build/* .; else true; fi"),
+                cwd=source_dir)
+        elif builddir != source_dir:
```

Comment:
Use `resolve()` when comparing paths, please:

```python
>>> Path('/etc/../etc') == Path('/etc/')
False
>>> Path('/etc/../etc').resolve() == Path('/etc/').resolve()
True
>>> 
```

(Maybe I should try adding a custom method for comparison, it's easy to fall for this trap...)

Reply:
1b30375e